tests/infra.py

```diff
@@ -79,3 +84,131 @@ def __getattr__(self, item: str) -> typing.Any:
             return super(DSSAsserts, self).__getattr__(item)  # type: ignore
         else:
             raise AttributeError(item)
+
+
+class S3TestBundle:
+    """
+    A test bundle staged in S3
+
+    This class does a little bit of "double duty" as we also use it to store the uuid and versions used with the API
+    """
+    BUCKET_TEST_FIXTURES = get_env('DSS_S3_BUCKET_TEST_FIXTURES')
+
+    def __init__(self, path, bucket=BUCKET_TEST_FIXTURES):
+        self.bucket = boto3.resource('s3').Bucket(bucket)
+        self.path = path
+        self.files = self.enumerate_bundle_files()
+        self.uuid = str(uuid.uuid4())
+        self.version = None
+
+    def enumerate_bundle_files(self):
+        object_summaries = self.bucket.objects.filter(Prefix=f"{self.path}/")
+        return [S3File(objectSummary, self) for objectSummary in object_summaries]
+
+
+class S3File:
+    """
+    A test file staged in S3
+    """
+
+    def __init__(self, object_summary, bundle):
+        self.bundle = bundle
+        self.path = object_summary.key
+        self.metadata = object_summary.Object().metadata
+        self.indexed = True if self.metadata['hca-dss-content-type'] == "application/json" else False
+        self.name = os.path.basename(self.path)
+        self.url = f"s3://{bundle.bucket.name}/{self.path}"
+        self.uuid = str(uuid.uuid4())
+        self.version = None
+
+
+class StorageTestSupport(DSSAsserts):
```

Comment:
Thanks very much for investigating and a proposed resolution. I think it is best to not have the new class extend DSSAsserts, at the expense of declaring `self` as `Any`.
This is how I did it, let me know if you have other recommendations:
    def create_bundle(self: Any, bundle: S3TestBundle):
        response = self.assertPutResponse(